.github/workflows/sonar-quality-gate.yml

```
@@ -0,0 +1,71 @@
+        
+      - name: Run linting
+        run: npm run lint
+        
+      - name: Run tests with coverage
+        run: npm run coverage
+        
+      - name: Generate ESLint report
+        run: npm run lint -- --format json --output-file eslint-report.json || true
+        
+      - name: Cache SonarQube packages
+        uses: actions/cache@v3
```

actionlint (1.7.7)

the runner of "actions/cache@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)